### PR TITLE
Use 'six' module to make the software Python3-compatible. Automatical…

### PR DESCRIPTION
**Pull request by PaxG**
Use 'six' module to make the software Python3-compatible. 
Automatically creates slow mountpoint if directory does not exist yet. 
Added -d shortcut for --debug. 
Code to be tested under Python2

### CODE REVIEW

File: slowfs.py [hunk 1]
```diff
@@ -15,14 +15,19 @@
 import sys
 import threading
 import time
-
-import fuse
-
+import fuse      # Filesystem in UserSpacE
+import six       # Python 2 and 3 Compatibility Library (2*3=6)
```
Comment (nirs) [inline slowfs.py]: I don't think these comments are needed, and they are not relevant for supporting python 3.

Also aligning the comments is not compatible with pep8. While we don't enforce pep8 yet, I don't like to introduce incompatible code.

Please check your changes with flake8  to make sure new code conforms to pep8.

File: slowfs.py [hunk 2]
```diff
@@ -15,14 +15,19 @@
...
 def main(args):
     args = parse_args(args)
     logging.basicConfig(level=logging.DEBUG if args.debug else logging.INFO,
                         format='%(asctime)s %(levelname)s [%(name)s] %(message)s')
+    if not os.path.exists(args.mountpoint):
+        try:
+            os.mkdir(args.mountpoint)
+        except Exception as e:
+            print("Error creating mountpoint {}: {}".format(args.mountpoint, e))
+            sys.exit(1)
```
Comment (nirs) [inline slowfs.py]: This change is not related to supporting python 3, and I don't think it is needed. This may create the mountpoint with incorrect ownership and permissions.

Please send this change in another PR if you like.

File: slowfs.py [hunk 3]
```diff
@@ -33,7 +38,7 @@ def parse_args(args):
     parser = argparse.ArgumentParser(description='A slow filesystem.')
     parser.add_argument('-c', '--config',
                         help='path to configuration file')
-    parser.add_argument('--debug', action='store_true',
+    parser.add_argument('-d', '--debug', action='store_true',
```
Comment (nirs) [inline slowfs.py]: Good change, but also not related to python 3 - please send another PR for rthis.

File: slowfs.py [hunk 4]
```diff
@@ -64,7 +69,9 @@ def set(self, name, value):
     def _load(self):
         d = {}
         with open(self._path) as f:
-            exec f in d, d
+            # exec f in d, d   # Not Python3 compatible syntax
+            # f is now a io.TextIOWrapper object, whereas "exec" needs a string, bytes or code object => add .read() method
+            six.exec_(f.read(),d)   # Execute code in the scope of globals and locals. code can be a string or a code object.
```
Comment (nirs) [inline slowfs.py]: Using six.exec_ is fine and should work on both python 2 and 3.

However using open() is not correct, we should use:

    io.open(self._path, "rb")

Which behave exactly the same in python 2 and 3, opening the file in binary mode.

Again the commented code should be removed, and the comment is not needed and should not be aligned in this way. If you think this needs some documentation, please add a docstring to the function.

File: slowfs.py [hunk 5]
```diff
@@ -98,7 +105,8 @@ def _run(self):
     def _handle_command(self):
         try:
             msg, sender = self.sock.recvfrom(1024)
-        except socket.error, e:
+        # except socket.error, e:    Not Python3 compatible syntax
+        except socket.error as e:
```
Comment (nirs) [inline slowfs.py]: Looks good, please remove the commented code.

**Review by nirs — CHANGES_REQUESTED** [review]
Thanks Pascal!

I hope you can remove the unrelated changes, and send separate PRs for them.

If not, I can take the python 3 related changes from your PR.

**Comment by PaxG** [comment]
Thanks for your review! I removed changes not related to Python 2/3 compatibility, removed useless comments and used flake8 to check code style. New changes are pushed on the python3 branch.

**Review by nirs — APPROVED** [review]
The patch looks fine now, thanks!

**Comment by nirs** [comment]
Thanks Pascal, I merge your PR with minor changes in 9e835eacc0e8596d725f9c1ac55072ae7f24b58e.

- Separate standard library modules (e.g. io) from 3rd party modules (e.g. six)
- Rewrite the commit message
